### ISSUE TITLE
Bash from msys2 in Docker on Windows doesn't understand volume mounts

### ISSUE THREAD

**Issue by jasharpe** [docker/for-win#1993]
### Steps to reproduce the behavior

Code to repro in https://gist.github.com/jasharpe/cc273b624440a49a1ec1f5db15e470ea. (Assumes git is installed. If not, the files from the gist can be copied manually.)

```
powershell
git clone https://gist.github.com/cc273b624440a49a1ec1f5db15e470ea.git
cd cc273b624440a49a1ec1f5db15e470ea
.\repro.ps1
```

It should result in the invocation of `bad.sh` in the container printing something like:

```
ls: cannot access '/c/work/': No such file or directory
total 404
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 May  2 17:46 .
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Oct 25  2016 ..
-rw-r--r-- 1 ContainerAdministrator ContainerAdministrator      1 Sep 29  2017 BOOTNXT
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Apr  2 20:15 Boot
lrwxrwxrwx 1 ContainerAdministrator ContainerAdministrator      8 Sep 29  2017 Documents and Settings -> /c/Users
-rw-r--r-- 1 ContainerAdministrator ContainerAdministrator   1894 Sep 29  2017 License.txt
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Apr  2 20:11 PerfLogs
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 May  2 17:42 Program Files
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Sep 29  2017 Program Files (x86)
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Apr  2 20:16 ProgramData
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 May  2 17:42 TEMP
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Apr  2 20:17 Users
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 May  2 17:46 WcSandboxState
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 Apr  2 20:16 Windows
-r--r--r-- 1 ContainerAdministrator ContainerAdministrator 398144 Apr  2 20:10 bootmgr
drwxr-xr-x 1 ContainerAdministrator ContainerAdministrator      0 May  2 17:43 tools
lrwxrwxrwx 1 ContainerAdministrator ContainerAdministrator     66 May  2 17:46 work -> /c/ContainerMappedDirectories/5E0F8EB0-A421-40CE-809E-5FE6F4A67701
``` 

Note the error on the first line. (The rest is included for context.)

### Expected behavior

`ls` of /c/work/ should succeed.

### Actual behavior

```
ls: cannot access '/c/work/': No such file or directory
```

### Information

Windows Server 2016 1706.

Basically what seems to be happening is that msys2 bash doesn't understand volume mounts in Docker. From the end of the output we see:

```
work -> /c/ContainerMappedDirectories/5E0F8EB0-A421-40CE-809E-5FE6F4A67701
```

But this "ContainerMappedDirectories" directory doesn't seem to actually exist anywhere, and even calling ls on this directly in bash doesn't work.
